Gems/Atom/Feature/Common/Code/Source/Debug/RenderDebugSettings.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Serialization/SerializeContext.h>
+
+#include <Atom/RPI.Public/RPIUtils.h>
+#include <Atom/RPI.Public/Shader/ShaderResourceGroup.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/View.h>
+
+#include <Debug/RenderDebugSettings.h>
+#include <Debug/RenderDebugFeatureProcessor.h>
+
+namespace AZ {
+    namespace Render {
+
+        RenderDebugSettings::RenderDebugSettings(RenderDebugFeatureProcessor* featureProcessor)
+            : m_featureProcessor(featureProcessor)
+        {
+        }
+
+        void RenderDebugSettings::Simulate()
+        {
+            UpdateOptionsMask();
+        }
+
+        void RenderDebugSettings::UpdateOptionsMask()
+        {
+            m_optionsMask = 0;
+
+            // Enabled
+            m_optionsMask = m_optionsMask | ((u32)GetEnabled() << (u32)RenderDebugOptions::DebugEnabled);
+
+            // Material Overrides
+            m_optionsMask = m_optionsMask | ((u32)GetOverrideBaseColor() << (u32)RenderDebugOptions::OverrideBaseColor);
+            m_optionsMask = m_optionsMask | ((u32)GetOverrideRoughness() << (u32)RenderDebugOptions::OverrideRoughness);
+            m_optionsMask = m_optionsMask | ((u32)GetOverrideMetallic() << (u32)RenderDebugOptions::OverrideMetallic);
```

Comment:
Why use this odd form?
`optionMask = optionMask | newBitFlag`
rather than 
`optionMask |= newBitFlag`

Reply:
Good call, will fix